src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -634,15 +678,104 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
          wtaq0(p)  = 0._r8
          obuold(p) = 0._r8
          btran(p)  = btran0
+         dhsdt_canopy(p) = 0._r8
+         eflx_sh_stem(p) = 0._r8
       end do
-      frac_rad_abs_by_stem = 0._r8
-      lw_leaf              = 0._r8
-      cp_leaf              = 0._r8
-      lw_stem              = 0._r8
-      sa_internal          = 0._r8
-      dt_stem              = 0._r8
-      tl_ini               = 0._r8
-      ts_ini               = 0._r8
+      !
+      ! Calculate biomass heat capacities
+      !
+      if(use_biomass_heat_storage) then
```

Comment:
My impression is that for this to work with FATES, we would split the rest of this clause for use_fates and use_cn, where each has its own subroutine that returns patch level 
sa_leaf(p)  frac_rad_abs_by_stem(p)  sa_stem(p) sa_internal(p) cp_leaf(p)  cp_stem(p) rstem(p)

And the subroutine for the fates side would probably live in clmfates_interfaceMod.F90

Reply:
No, really we just want FATES to provide stem_biomass and leaf_biomass. There's an if inside this section you outline for either BGC or SP that will possibly need a use_fates option to provide those two. 

It's actually probably better if those two were provided for them outside of this so that CanopyFluxes doesn't have to know if FATES or BGC or SP are running, it just uses the biomass provided into it.

I think this should be a simple refactoring we do later.

I've now got an issue for this here: #1247

Reply:
I'm going to resolve this since I've moved it to the issue.